### ISSUE TITLE
Have pay just pay a bolt12 offer

### ISSUE THREAD

**Issue by Sjors** [ElementsProject/lightning#7632]
I ran into the same confusion that led to #5473.

```
lightning-cli pay lno1zrxq... 100sat
{
   "code": -32602,
   "message": "Invalid bolt12: unexpected prefix lno"
}
```

You have to call `fetchinvoice` and then `pay` that.

Why not just do that automatically?

**Comment by vincenzopalazzo** [comment]
>You have to call fetchinvoice and then pay that.

This is true for any bolt12. It is also true that it is trivial to add a `pay <bolt12>` helper

**Comment by Sjors** [comment]
I can imagine needing a different flow for `bolt12` offers that give away money. And repeating payments also aren't very suitable for the `pay` call.

**Comment by vincenzopalazzo** [comment]
>I can imagine needing a different flow for bolt12 offers that give away money. And repeating payments also aren't very suitable for the pay call.

You are right, in fact I have a plugin that wrap the bolt12, so it is time to upstream the changes. Thanks for reporting

**Comment by rustyrussell** [comment]
Currency conversions and repeated payments both need a different flow, so it's generalized.  But if the amount is known and it's not repeated (which, TBF, is very common), it's totally possible to have pay Just Work,

**Comment by Sjors** [comment]
Could currency conversion be handled through a plugin? So you'd simply use `pay bolt12_fiat_amount` and if the plugin is there, it just works, and otherwise it complains "sorry, no plugin".

**Comment by vincenzopalazzo** [comment]
>Could currency conversion be handled through a plugin? So you'd simply use pay bolt12_fiat_amount and if the plugin is there, it just works, and otherwise it complains "sorry, no plugin".

Yes, but this is another story, the pay plugin should handle bolt12 IMHO, I will try to see what kind of code is required to implement it

**Comment by rustyrussell** [comment]
I'm moving this to next milestone, sorry.  I'd really like xpay to do this at some point though...
